pypeit/spectrographs/spectrograph.py

```diff
@@ -514,7 +516,46 @@ def get_rawimage(self, raw_file, det):
                 oscansec_img = pix_img.copy()
 
         # Return
-        return raw_img, hdu, exptime, rawdatasec_img, oscansec_img
+        return raw_img, headarr, exptime, rawdatasec_img, oscansec_img
+
+    def get_lamps_status(self, headarr):
+        """
+        Return a string containing the information on the lamp status
+
+        Args:
+            headarr (list of fits headers):
+              list of headers
+
+        Returns:
+            str: A string that uniquely represents the lamp status
+        """
+        # Loop through all lamps and collect their status
+        kk = 1
+        lampstat = []
+        while True:
+            lampkey = 'lampstat{:02d}'.format(kk)
+            if lampkey not in self.meta.keys():
+                break
+            ext = self.meta[lampkey]['ext']
+            card = self.meta[lampkey]['card']
+            lampstat += [str(headarr[ext][card])]
+            kk += 1
+        return "_".join(lampstat)
+
+    def get_arclamps(self, filename):
```

Comment:
This is an odd method.  Should it be removed?

Reply:
This is a dummy method in the `spectrograph.py` file, but I use this method in `keck_kcwi.py` to grab the lamps. I remember convincing myself that I can't do the same with the `config_specific_par` routine, but it was so long ago... let me investigate, and report back...

Reply:
Righto - there are two possible arc lamps in KCWI - FeAr and ThAr. This routine is designed to get the information from the header about what lamps are currently on. The alternative is to force the user to set the lamps manually in their `.pypeit` file.

So, I can delete this routine if you'd prefer the manual option. What do you think?

Reply:
OK - as per discussion, I've removed this code... I've commented out the code in `keck_kcwi.py`, so that we can use that when getting the metadata.